lib/rpmts.c

```diff
@@ -615,6 +616,16 @@ rpmRC rpmtsImportPubkey(const rpmts ts, const unsigned char * pkt, size_t pktlen
     if (txn == NULL)
 	return rc;
 
+    krc = pgpPubKeyLint(pkt, pktlen, &lints);
+    if (lints) {
+	fprintf(stderr, "%s\n", lints);
```

Comment:
We can't really use fprintf() from librpm because it's a PITA for API users.

Rpm generally follows the "no news is good news" principle, and while enabling extra output with `-v` (for verbose) would fit, RPMLOG_INFO and NOTICE end up outputting to stdout which is a different kind of pain. I'd suggest using rpmlog() with log level based on `krc` - on error use RPMLOG_ERR and otherwise RPMLOG_DEBUG, that's the common workaround.

Reply:
It wasn't clear to me from the sample output + code that the lint messages are always warnings. If that is the case then RPMLOG_WARNING is the right thing to do, but we don't want "considering/linting key foo" style messages visible by default.

Reply:
> we don't want "considering/linting key foo" style messages visible 

Sorry about that, the "Consider..." lines are local debug output that I forgot to remove.  (I was confused about why I wasn't seeing any output, before I realized I needed two -vv's.)

> It wasn't clear to me from the sample output + code that the lint messages are always warnings. 

There are two cases where I generate lints: for unusable certificates, and suspicious certificates.  A suspicious certificate might be one where the primary key is okay, but a signing-capable subkey uses SHA-1 and is thus considered invalid.

Reply:
A subkey that uses SHA-1 is ignored, but that doesn't mean the primary key can't be used to verify signatures.  That's why that case is a warning and not an error.

Reply:
Sure, in that case RPMLOG_WARNING/ERROR, depending on krc is the right thing.

Reply:
As far as I can tell, `RPMLOG_WARNING` log messages are only shown when `-vv` is passed to `rpm`.  Doesn't that mean in practice few people will actually see those warnings?

Reply:
WARNING and ERROR level messages are practically always output, because, well, people need to see them. It's the levels above those that are dependent on -v/-vv.